core/src/main/java/com/linecorp/armeria/common/stream/FixedStreamMessage.java

```diff
@@ -33,277 +36,465 @@
  * Reduced synchronization and allocation allow for much higher performance than {@link DefaultStreamMessage},
  * so this class should generally be used when the objects are known.
  */
-public class FixedStreamMessage<T> extends AbstractStreamMessage<T> {
+public final class FixedStreamMessage {
```

Comment:
How about:

- moving all the factory methods to `StreamMessage`
- removing `FixedStreamMessage` and moving all its nested classes top-level
- renaming `AbstractFixedStreamMessage` to `FixedStreamMessage`?

Reply:
Thanks - reorganized. Unfortunately I can't make the `FixedStreamMessage` implementations package-private since we inherit from outside the package. It's why I was thinking of moving them to `internal` but I guess this way is ok for now.